confdgnmi/src/confd_gnmi_server.py

```diff
@@ -57,6 +56,52 @@ def get_and_connect_adapter(self, username, password):
         log.debug("<== adapter=%s", adapter)
         return adapter
 
+    def get_val_encoding(self, val):
+        log.debug("==> val=%s", val)
+        encoding = None
+        if str(val).startswith("json_val"):
```

Comment:
`str_val = str(val)` before the if/else chain, so you don't have many `str(val)` repetitions...